Rewrite to /bin/sh, add --config option

slotThe opened the pull request:
This does a few things, but I thought it would be cleaner to do this
than to clutter things up with different PRs.  It is a bit of a mix
between cleanup and features, though the commits are reasonably atomic
at least.

Overview:

- Add magit-section to dependencies

  This is needed on some systems ever since the magit v3 release.  If
  this is not present, magit.sh will error out saying that it can't find
  the magit-section package.

- Rewrite magit.sh to use /bin/sh

  This increases portability of the script and makes it effortlessly
  available even on systems that do not have a /bin/bash.  It does add a
  dependency on sed in order to recreate `printf '%q'` but, given the
  nature of the arguments, that may well be removed—printf '%s' is just
  fine in our case.

  Fixes: https://github.com/alphapapa/magit.sh/issues/2

- Shift inside argument case statement

  Instead of shifting outside of the case and always looping, actually
  check if there is something to loop over.  This allows one to have
  much more freedom with regards to options with different arguments.

- Add -c/--config option

  This option allows for configuring the Emacs configuration path; i.e.,
  the path that we look for when it comes to sourcing packages.

slotThe committed:
Add magit-section to dependencies

This is needed on some systems ever since the magit v3 release.  If
this is not present, magit.sh will error out saying that it can't find
the magit-section package.

alphapapa commented:
Thanks, but while adding magit-section to the dependencies is obviously needed, these other changes aren't.  I don't want to reduce the script to mere POSIX shell--Bash is a better scripting language.

Regardless, since I made this repo, I made https://github.com/alphapapa/emacs-sandbox.sh, which is probably a better solution than this script, anyway.  e.g. `emacs-sandbox.sh -d ~/.cache/emacs/sandboxes/magit -i magit` to initialize, then `emacs-sandbox.sh -d ~/.cache/emacs/sandboxes/magit -R` to run it normally.  That way, it's independent of the user's primary Emacs config, and there are other improvements as well.

So rather than merge this PR, it'd probably be better to rewrite the readme to point users at `emacs-sandbox.sh` and show how to use it to accomplish the same thing.

slotThe committed:
Change shebang to /usr/bin/env bash

Some distributions, like NixOS, and other operating systems, like many
BSDs, do not have a /bin/bash.  This uses the env-based invocation of
bash to work around this.

Fixes: https://github.com/alphapapa/magit.sh/issues/2

slotThe commented:
> Thanks, but while adding magit-section to the dependencies is obviously needed, these other changes aren't. I don't want to reduce the script to mere POSIX shell--Bash is a better scripting language.

Fair enough.

> Regardless, since I made this repo, I made https://github.com/alphapapa/emacs-sandbox.sh, which is probably a better solution than this script, anyway. e.g. `emacs-sandbox.sh -d ~/.cache/emacs/sandboxes/magit -i magit` to initialize, then `emacs-sandbox.sh -d ~/.cache/emacs/sandboxes/magit -R` to run it normally. That way, it's independent of the user's primary Emacs config, and there are other improvements as well.
> 
> So rather than merge this PR, it'd probably be better to rewrite the readme to point users at `emacs-sandbox.sh` and show how to use it to accomplish the same thing.

This is already mentioned in the current README, so I don't think much still has to be done.  I have to say that I like using `magit.sh` specifically because it re-uses already installed packages and doesn't install them twice.  

I have updated the PR to only include adding `magit-section`, as well as changing the shebang to `/usr/bin/env bash` (so that this still addresses #2); what do you think?

alphapapa commented:
> This is already mentioned in the current README, so I don't think much still has to be done. 

Haha, I had forgotten that I did that about 15 months ago.  At least I know I still agree with my past self.  :)

> 
> I have updated the PR to only include adding `magit-section`, as well as changing the shebang to `/usr/bin/env bash` (so that this still addresses #2); what do you think?

Looks good to me, thanks.
